### DISCUSSION TITLE
Long loading times when offline.

### DISCUSSION THREAD

**Discussion by SoffiMeower** [ppy/osu#21241] [Q&A] [answered]
### Type

Game behaviour

### Bug description

When offline, osu!lazer typically takes an extremely long time (30+ secs) to load a beatmap. (Possibly trying to contact server?) Leading to a poor user experience when playing osu!lazer while offline. Being offline is pretty common for mobile users, especially on the go. While for desktop users, they could have a network outage and want to play osu! to pass the time, except they're mostly met with delayed loading. This could be mitigated by letting lazer passively attempt to reconnect in the background, while letting the user just keep their scores localized.

### Screenshots or videos

A comparison video, the 1st launch is with my ethernet plugged in, and the 2nd launch is with the ethernet unplugged from my desktop. (Total disconnect).

https://user-images.githubusercontent.com/45863583/201758058-cd6e1770-2249-45c3-be14-75f6bf3988bf.mp4

Connected: Approx. 2 seconds.
Disconnected: Approx. 1 minute and 1 second

### Version

2022.1101.0-lazer

### Logs

[input.log](https://github.com/ppy/osu/files/10003385/input.log)
[network.log](https://github.com/ppy/osu/files/10003387/network.log)
[performance.log](https://github.com/ppy/osu/files/10003389/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10003390/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10003391/updater.log)
[database.log](https://github.com/ppy/osu/files/10003392/database.log)

**Comment by peppy** [comment]
The logs show that you are getting an actual web response durin the disconnected time, which means something on your network or PC is intercepting the osu! requests and providing an unexpected response.

Can you try "disconnecting" your internet again and then visiting the osu! site and seeing what you see?

**Reply by SoffiMeower** [reply]
Let me try again when home. This was when I was on break during work, so it probably was getting a half-reply. OBS recording was from my PC when I was home and unplugged the cord. I'll send logs for my desktop when I get home.

**Comment by SoffiMeower** [comment]
Trying a mobile or cut-out situation: (Connected at first, but then disconnected. Has long loading.)
[updater.log](https://github.com/ppy/osu/files/10017844/updater.log)
[database.log](https://github.com/ppy/osu/files/10017845/database.log)
[input.log](https://github.com/ppy/osu/files/10017846/input.log)
[network.log](https://github.com/ppy/osu/files/10017847/network.log)
[runtime.log](https://github.com/ppy/osu/files/10017848/runtime.log)
[performance.log](https://github.com/ppy/osu/files/10017849/performance.log)

**Comment by SoffiMeower** [comment]
A completely disconnected situation (Network completely disabled.  Has long loading.)
[network.log](https://github.com/ppy/osu/files/10017859/network.log)
[runtime.log](https://github.com/ppy/osu/files/10017860/runtime.log)
[performance.log](https://github.com/ppy/osu/files/10017861/performance.log)
[updater.log](https://github.com/ppy/osu/files/10017862/updater.log)
[database.log](https://github.com/ppy/osu/files/10017863/database.log)
[input.log](https://github.com/ppy/osu/files/10017864/input.log)

**Comment by Doksha** [comment]
this always happens when you are trying to load a map while the client is trying to log in (ie no network) instead of logging out of the account. you just need to cancel the login (yeah its annoying having to log in each time you get back online) still should be fixed though.

this does not happen if lazer isnt trying to log in (ie you are logged out)

**Reply by SoffiMeower** [reply]
Yes, but it should atleast be able to detect a connection being missing. *Either by asking the OS the connection status, or giving a ping to the osu.ppy.sh website*

**Comment by peppy** [comment, marked as answer]
#21288
